File: apps/api/src/routes/siteContent.ts
import { Router, Request, Response } from 'express';
import { requireAdmin } from '../middleware/auth';
import { SiteContentRepository } from '../repositories/SiteContentRepository';

const router = Router();

// GET /api/site-content — public
router.get('/', (_req: Request, res: Response) => {
  res.json(SiteContentRepository.get());
});

// PATCH /api/site-content — admin only
router.patch('/', requireAdmin, (req: Request, res: Response) => {
  const { tagline, description, highlights, instagram_handle } = req.body;
  const errors: string[] = [];

  if (tagline !== undefined) {
    if (typeof tagline !== 'string' || tagline.trim().length === 0) {
      errors.push('tagline must be a non-empty string');
    } else if (tagline.length > 150) {
      errors.push('tagline must be 150 characters or fewer');
    }
  }

  if (description !== undefined) {
    if (typeof description !== 'string') {
      errors.push('description must be a string');
    } else if (description.length > 500) {
      errors.push('description must be 500 characters or fewer');
    }
  }

  if (highlights !== undefined) {
    if (!Array.isArray(highlights)) {
      errors.push('highlights must be an array');
    } else {
      highlights.forEach((h: unknown, i: number) => {
        if (typeof h !== 'string') {
          errors.push(`highlights[${i}] must be a string`);
        } else if (h.length > 80) {
          errors.push(`highlights[${i}] must be 80 characters or fewer`);
        }
      });
    }
  }

  if (errors.length > 0) {
    res.status(400).json({ errors });
    return;
  }

  const partial: any = {};
  if (tagline !== undefined) partial.tagline = tagline.trim();
  if (description !== undefined) partial.description = description;
  if (highlights !== undefined) partial.highlights = highlights;
  if (instagram_handle !== undefined) partial.instagram_handle = instagram_handle || null;

  const updated = SiteContentRepository.update(partial);
  res.json(updated);
});

export default router;

File: apps/api/routers/site_content.py
from fastapi import APIRouter, Depends, HTTPException

from middleware.auth import require_admin
from models.domain import PatchSiteContentPayload, SiteContent
from repositories import site_content_repo

router = APIRouter(prefix="/site-content", tags=["site-content"])


@router.get("/", response_model=SiteContent)
def get_site_content():
    return site_content_repo.get()


@router.patch("/", response_model=SiteContent)
def update_site_content(
    payload: PatchSiteContentPayload,
    _: dict = Depends(require_admin),
):
    errors = []
    if payload.tagline is not None:
        if not payload.tagline.strip():
            errors.append("tagline must be a non-empty string")
        elif len(payload.tagline) > 150:
            errors.append("tagline must be 150 characters or fewer")
    if payload.description is not None and len(payload.description) > 500:
        errors.append("description must be 500 characters or fewer")
    if payload.highlights is not None:
        for i, h in enumerate(payload.highlights):
            if len(h) > 80:
                errors.append(f"highlights[{i}] must be 80 characters or fewer")
    if errors:
        raise HTTPException(status_code=400, detail={"errors": errors})

    partial = payload.model_dump(exclude_unset=True)
    if "tagline" in partial:
        partial["tagline"] = partial["tagline"].strip()
    return site_content_repo.update(partial)
